File: client-app/src/app/models/crelic.ts
export interface Crelic {
    id: string
    name: string
    charges: number
    isHidden: boolean
    maxCharges: number
    crelicAbilities: Array<CrelicAbility>
}

export interface CrelicAbility {
    id: string
    name: string
    description: string
    crelicSubAbilities: Array<CrelicSubAbility>
}

export interface CrelicSubAbility {
    id: string
    name: string
    level: number
    crelicSubAbilityLevels: Array<CrelicSubAbilityLevel>
}

export interface CrelicSubAbilityLevel {
    id: string
    description: string
    level: number
    cost: number
}

File: Domain/crelics/Crelic.cs
namespace Domain
{
    public class Crelic
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public int Charges { get; set; } = 10;
        public int MaxCharges { get; set; } = 10;
        public bool isHidden { get; set; } = true;
        public ICollection<CrelicAbility> CrelicAbilities { get; set; } = new List<CrelicAbility>();
    }
}